src/main/java/org/hoverla/bibernate/connectionpool/util/DriverDataSource.java

```diff
@@ -0,0 +1,149 @@
+package org.hoverla.bibernate.connectionpool.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.sql.DataSource;
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+
+/**
+ * Utility data source to get low-level connections to filling the pool.
+ */
+@Slf4j
+public class DriverDataSource implements DataSource {
+
+    private static final String PASSWORD = "password";
+    private static final String USER = "user";
+
+    private final String jdbcUrl;
+    private final Properties driverProperties;
+    private Driver driver;
+
+    public DriverDataSource(String jdbcUrl, String driverClassName, String username, String password) {
+        this.jdbcUrl = jdbcUrl;
+        this.driverProperties = new Properties();
+
+        if (username != null) {
+            driverProperties.put(USER, driverProperties.getProperty(USER, username));
+        }
+        if (password != null) {
+            driverProperties.put(PASSWORD, driverProperties.getProperty(PASSWORD, password));
+        }
+
+        if (driverClassName != null) {
+            var drivers = DriverManager.getDrivers();
+            while (drivers.hasMoreElements()) {
+                var d = drivers.nextElement();
```

Comment:
"d" is an unclear name.

Reply:
Also, this logic should not write in the constructor. Create separate methods for impl this logic and add dosc ,please.